File: scripts/generate-icons.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const ICONS_DIR = path.join(__dirname, '..', 'src-tauri', 'icons');

// Create icons directory if it doesn't exist
if (!fs.existsSync(ICONS_DIR)) {
  fs.mkdirSync(ICONS_DIR, { recursive: true });
}

const SOURCE_LOGO = path.join(__dirname, '..', 'src-tauri', 'icons', 'logo.png');

// Generate different size PNGs
const sizes = [
  { size: 32, name: '32x32.png' },
  { size: 128, name: '128x128.png' },
  { size: 256, name: '128x128@2x.png' },
];

async function generateIcons() {
  // Generate PNG icons
  for (const { size, name } of sizes) {
    await sharp(SOURCE_LOGO)
      .resize(size, size)
      .png()
      .toFile(path.join(ICONS_DIR, name));
  }

  // Generate Windows ICO
  await sharp(SOURCE_LOGO)
    .resize(256, 256)
    .toFormat('ico')
    .toFile(path.join(ICONS_DIR, 'icon.ico'));

  console.log('Icons generated successfully!');
}

generateIcons().catch(console.error);

File: scripts/generate_icons.py
from PIL import Image
import os

# Set up paths
base_dir = os.path.dirname(os.path.dirname(__file__))
icons_dir = os.path.join(base_dir, "src-tauri", "icons")
source_logo = os.path.join(icons_dir, "logo.png")

# Create icons directory if it doesn't exist
os.makedirs(icons_dir, exist_ok=True)

# Load source image
img = Image.open(source_logo)

# Generate PNG icons
sizes = [(32, "32x32.png"), (128, "128x128.png"), (256, "128x128@2x.png")]

for size, filename in sizes:
    resized = img.resize((size, size), Image.Resampling.LANCZOS)
    resized.save(os.path.join(icons_dir, filename), "PNG")

# Generate ICO file with multiple sizes
ico_sizes = [(16, 16), (32, 32), (48, 48), (256, 256)]
ico_images = []

for size in ico_sizes:
    ico_images.append(img.resize(size, Image.Resampling.LANCZOS))

ico_path = os.path.join(icons_dir, "icon.ico")
ico_images[0].save(
    ico_path,
    format="ICO",
    sizes=[(im.width, im.height) for im in ico_images],
    append_images=ico_images[1:],
)

print("Icons generated successfully!")
